<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>870</width>
    <height>667</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="CodeEdit" name="codeEdit" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-1</y>
      <width>460</width>
      <height>410</height>
     </rect>
    </property>
   </widget>
   <widget class="DrawArea" name="drawArea" native="true">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>0</y>
      <width>410</width>
      <height>410</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="consoleTitle">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>410</y>
      <width>870</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="text">
     <string> 控制台</string>
    </property>
   </widget>
   <widget class="TextConsole" name="console">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>440</y>
      <width>870</width>
      <height>130</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>870</width>
     <height>29</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <widget class="QMenu" name="file">
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="_new"/>
    <addaction name="_open"/>
    <addaction name="_save"/>
    <addaction name="_saveAs"/>
   </widget>
   <widget class="QMenu" name="edit">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="_undo"/>
    <addaction name="_redo"/>
    <addaction name="_cut"/>
    <addaction name="_copy"/>
    <addaction name="_paste"/>
   </widget>
   <widget class="QMenu" name="settings">
    <property name="title">
     <string>设置</string>
    </property>
    <addaction name="skinSettings"/>
    <addaction name="vmSettings"/>
   </widget>
   <addaction name="file"/>
   <addaction name="edit"/>
   <addaction name="settings"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="font">
    <font>
     <family>Microsoft YaHei UI</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="run"/>
   <addaction name="loadCommand"/>
   <addaction name="runByLine"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="_new">
   <property name="text">
    <string>新建</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="_open">
   <property name="text">
    <string>打开…</string>
   </property>
   <property name="toolTip">
    <string>打开…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="_save">
   <property name="text">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="_saveAs">
   <property name="text">
    <string>另存为…</string>
   </property>
   <property name="toolTip">
    <string>另存为…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="_undo">
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="_redo">
   <property name="text">
    <string>重做</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="_cut">
   <property name="text">
    <string>剪切</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="_copy">
   <property name="text">
    <string>复制</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="_paste">
   <property name="text">
    <string>粘贴</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="_settings">
   <property name="text">
    <string>设置</string>
   </property>
  </action>
  <action name="run">
   <property name="text">
    <string>运行</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="loadCommand">
   <property name="text">
    <string>加载指令</string>
   </property>
   <property name="toolTip">
    <string>把文件中的指令加载进虚拟机但不运行，并启动控制台</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F5</string>
   </property>
  </action>
  <action name="skinSettings">
   <property name="text">
    <string>外观设置...</string>
   </property>
  </action>
  <action name="vmSettings">
   <property name="text">
    <string>虚拟机设置...</string>
   </property>
  </action>
  <action name="runByLine">
   <property name="text">
    <string>逐行运行…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+F5</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CodeEdit</class>
   <extends>QWidget</extends>
   <header location="global">qt_main/codeedit.h</header>
   <container>1</container>
   <slots>
    <slot>cut()</slot>
    <slot>copy()</slot>
    <slot>paste()</slot>
    <slot>undo()</slot>
    <slot>redo()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>DrawArea</class>
   <extends>QWidget</extends>
   <header location="global">qt_main/drawarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TextConsole</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">qt_main/textconsole.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>_cut</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_copy</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_paste</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_undo</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_redo</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
